tests/parametric/test_span_links.py

```diff
@@ -0,0 +1,189 @@
+import pytest
+
+from ddapm_test_agent.trace import Span
+from ddapm_test_agent.trace import Trace
+from ddapm_test_agent.trace import root_span
+from utils.parametric.spec.trace import SAMPLING_PRIORITY_KEY, ORIGIN
+from utils.parametric.spec.trace import find_span
+from utils.parametric.spec.trace import find_trace_by_root
+from utils.parametric.spec.trace import span_has_no_parent
+from utils.parametric.headers import make_single_request_and_get_inject_headers
+from utils.parametric.test_agent import get_span
+from utils import bug, context, scenarios
+from utils.parametric._library_client import Link
+
+
+@scenarios.parametric
+class Test_Span_Links:
+    def test_span_started_with_link(self, test_agent, test_library):
+        """Test adding a span link created from another span.
+        This tests the functionality of "create a direct link between two spans
+        given two valid span (or SpanContext) objects" as specified in the RFC.
+        """
+        with test_library:
+            with test_library.start_span("root") as parent:
+                with test_library.start_span(
+                    "child",
+                    parent_id=parent.span_id,
+                    links=[Link(parent_id=parent.span_id, attributes={"foo": "bar", "array": ["a", "b", "c"]})],
+                ):
+                    pass
+
+        traces = test_agent.wait_for_num_traces(1)
+        trace = find_trace_by_root(traces, Span(name="root"))
+        assert len(trace) == 2
+
+        root = root_span(trace)
+        span = find_span(trace, Span(name="child"))
+
+        assert span.get("parent_id") == root.get("span_id")
+        assert span.get("span_links") is not None
+
+        span_links = span["span_links"]
+        assert len(span_links) == 1
+        link = span_links[0]
+        assert link.get("span_id") == root.get("span_id")
+        assert link.get("trace_id") == root.get("trace_id")
+        assert (link.get("trace_id_high") or 0) == 0
```

Comment:
Where does the `trace_id_high` is coming from and what is its purpose?

Reply:
With 128-bit turned on by default, this will break. Perhaps on the right side of the equation we should be getting the span's `_dd.p.tid` tag value

Reply:
Good point, I need to change that ... on to figuring our how to convert between hex and decimal in python :-D